RentACarWebAPI/Startup.cs

```diff
@@ -41,6 +43,10 @@ public void ConfigureServices(IServiceCollection services)
             services.AddTransient<IRepositoryHelper<Car>, RepositoryHelper<Car>>();
             services.AddTransient<IRepositoryHelper<Client>, RepositoryHelper<Client>>();
             services.AddTransient<IRepositoryHelper<Rental>, RepositoryHelper<Rental>>();
```

Comment:
Is this a good approach? or should i change architecture to implement dependency injection to the Repository Helper?

Reply:
Good question. I think this is kinda fine because of the way you did it but IMHO I think I prefer creating explicit repository classes instead... even if that means having them empty (without the need of overwriting any method).

Example:

```
services.AddTransient<IRepositoryHelper<Car>, CarRepository>();
services.AddTransient<IRepositoryHelper<Client>, ClientRepository>();
services.AddTransient<IRepositoryHelper<Rental>, RentalRepository>();
```

Reply:
Ahá! im a bit confused, but let me take a look and if i can't understand it, i'll ask you! Thank you Mati!

Reply:
Keep in mind that each repository class will inherit from `IRepositoryHelper<YourSpecificClass>`. If you go that route consider renaming that interface as well... `IRepository` reads better.

Reply:
Okey, im getting closer to understanding! hahah.. but wait, i already have an `IBaseRepository`, So should I do one of those two?

summarizing:
```
    public interface IBaseRepository<T> where T : Entity
    {
        T Create(T obj);
        T Get(int id);
        T Update(T obj, int id);
        void Delete(int id);
        List<T> GetAll();

        List<T> CheckFileAndGetList(string filePath);
        int GetNewId(List<T> objs);
        List<T> GetListFromFile(string filePath);
        void SaveListToFile(List<T> objs, string filePath);
    }
}
```

Reply:
Ah my bad... didn't knew you had all those classes already! I made some assumptions with the current code.

Let's start again. What you did is fine with the code you have. Sorry for detouring you.

Now, there's also an alternative. It's up to you if you want to explore it or not. In my head I would have made this Repository a `JsonRepository : IRepository<T>` that internally knows how to serialize and deserialize and make JSON happy. That way your remove your helper dependency. You resolve all those things inside your class with private methods.

I'm not saying one approach is better than the other, is just an alternative in case you have doubts about why you had to register these 3 helpers.

Reply:
its okey! Everything that makes me think comes fine! hahah..
Good! I understand perfect now! Thanks for your explanation and your time! :D